src/interpreter.c

```diff
@@ -888,10 +886,7 @@ jl_value_t *NOINLINE jl_interpret_toplevel_thunk(jl_module_t *m, jl_code_info_t
     s->mi = NULL;
     s->ci = NULL;
     JL_GC_ENABLEFRAME(s);
-    jl_task_t *ct = jl_current_task;
-    size_t last_age = ct->world_age;
     jl_value_t *r = eval_body(stmts, s, 0, 1);
-    ct->world_age = last_age;
```

Comment:
Who's responsible now for correcting the value on return from eval_body? Seems like that scoped reset is failing to occur now?

Reply:
The original idea was that the effect from :thunk would persist, but since i added back the toplevel effect, I think I can probably restore the scoped reset here.

Reply:
I think the challenge is that it cannot persist out from `eval(Expr(:latestworld))`, so we have to make sure there is at least some place it is getting reset on that path. Beyond that, it is just a QOL that we can assume that world is always a scoped value, so any variance from that seen in the debugger means something went wrong elsewhere